observer/observer.go

```diff
@@ -67,7 +70,16 @@ func (o *Observer) processBlock(events chan<- Event, block *blockatlas.Block) {
 					Decimals: coin.Coins[o.Coin].Decimals,
 				}
 			}
-			emitted[tx.ID] = true
+
+			if d, ok := emitted[tx.ID]; ok {
+				if d == tx.Direction || d == blockatlas.DirectionSelf {
```

Comment:
More simple for read:
```go
if (d != tx.Direction && d != blockatlas.DirectionSelf) {
  emitted[tx.ID] = blockatlas.DirectionSelf
}
```

Reply:
but we need to call the `continue` to loop go to the next object and don't send the event again